### PR TITLE
Refactor winningest and worst coach

### PR DESCRIPTION
**Pull request by s-steel**
Refactored winningest and worst coach methods.
Increased number of test assertions for season statistics methods.
All tests passing.

### CODE REVIEW

File: lib/season_stats_module.rb [hunk 1]
```diff
@@ -19,40 +13,28 @@ def season_coaches(season_id)
...
-
-      total_wins = game_teams_data_for_season(season_id).count do |game|
-        game.head_coach == coach && game.result == "WIN"
-      end
-      coaches_hash[coach] = ((total_wins.to_f/total_games.to_f) * 100).round(2)
-
-    end
-
-    worst_coach = coaches_hash.min_by do |coach, win_percentage|
+    coaches_by_win_percentage(season_id).min_by do |coach, win_percentage|
```
Comment (cpfergus1) [inline lib/season_stats_module.rb:37]: nice refactor!

File: test/season_stats_module_test.rb [hunk 2]
```diff
@@ -19,19 +19,20 @@ def setup
 
   end
 
-  def test_find_all_games_from_season
```
Comment (cpfergus1) [inline test/season_stats_module_test.rb:22]: Is this test now obsolete?

File: lib/season_stats_module.rb [hunk 3]
```diff
@@ -19,40 +13,28 @@ def season_coaches(season_id)
     end.uniq
   end
 
-  def winningest_coach(season_id)
```
Comment (cpfergus1) [inline lib/season_stats_module.rb:22]: Shouldn't we leave the naming conventions winningest_coach due to the project spec?

**Review by cpfergus1 — APPROVED** [review]

**Review by GregJMitchell — APPROVED** [review]
Great refactoring! Looks great!

File: lib/season_stats_module.rb [hunk 4]
```diff
@@ -1,12 +1,6 @@
 module SeasonStatistics
   require 'pry'
 
-  def find_all_games_from_season(season_id)
-    @games.find_all do |game|
-      game.season == season_id
-    end
-  end
```
Comment (c-bartell) [inline lib/season_stats_module.rb:8]: Good call to remove this.

**Review by c-bartell — APPROVED** [review]
Good to go!

**Pull request merged**